lib/foreman/exception.rb

```diff
@@ -55,6 +55,28 @@ def message
     end
   end
 
+  class MultiException < ::Foreman::Exception
+    def initialize(exceptions, message, *params)
+      super(message, *params)
+      raise "Can't create multi exception - not an array: #{exceptions}" unless exceptions.respond_to?(:to_a)
+      @exceptions = exceptions.to_a.dup.freeze
+    end
+
+    attr_reader :exceptions
+
+    def string_form
+      super if @wrapped_exception.empty?
+      super + @exceptions.map { |e| e.message }
+    end
+
+    def json_form
+      {
+        message: message,
+        errors: (@exceptions.map { |e| e.message })
+      }.as_json
+    end
```

Comment:
Please create `as_json` method in Exception base class and override here (call it and merge `errors` key into the hash).

Also I suggest to do something like: `e.respond_to?(:message) ? e.message : e.to_s`